cdap-ui/app/hydrator/controllers/create/leftpanel-ctrl.js

```diff
@@ -162,7 +162,9 @@ class HydratorPlusPlusLeftPanelCtrl {
       let connections = [];
       let i;
       for (i=0; i<nodes.length - 1 ; i++) {
-        connections.push({ from: nodes[i].name, to: nodes[i+1].name });
+        if(nodes[i] && nodes[i + 1]) {
+          connections.push({ from: nodes[i].name, to: nodes[i+1].name });
```

Comment:
When there are no connection in JSON file, linear connections are generated and this line throws error if source, transforms, and sinks are not present in config property.